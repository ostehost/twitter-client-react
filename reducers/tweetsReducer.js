import sortItems from '../lib/sortItems'

import actionTypes from '../actions/actionTypes';

export const initialState = {
    search: '',
    count: '10',
    type: false,
    tweets: [],
    tweets_loading: false,
    sort_by: 'created_at',
    sort_order: 1,
    error: null
}

const tweetsReducer = (state = initialState, action) => {
    switch (action.type) {
        case actionTypes.CHANGE_TYPE:
            return { ...state, type: action.value };
        case actionTypes.CHANGE_COUNT:
            return { ...state, count: action.value };
        case actionTypes.CHANGE_SEARCH:
            return { ...state, search: action.value };
        case actionTypes.LOAD_TWEETS:
            return { ...state, tweets: [], tweets_loading: true };
        case actionTypes.TWEETS_LOADED:
            let tweets = action.append ? [action.payload, ...state.tweets] : action.payload;
            sortItems(tweets, state.sort_by, state.sort_order);
            return { ...state, error: null, tweets: tweets, tweets_loading: false };
        case actionTypes.SORT:
            sortItems(state.tweets, action.sortBy, action.sortOrder);
            return { ...state, sort_by: action.sortBy, sort_order: action.sortOrder }
        case actionTypes.LOAD_TWEETS_FAILURE:
            return {
                ...state,
                error: action.error,
                tweets_loading: false
            }
        default:
            return state
    }
}

export default tweetsReducer;