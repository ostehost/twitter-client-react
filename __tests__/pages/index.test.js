/* eslint-env jest */
import { render } from '@testing-library/react'

import App from '../../pages/index.js'

import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

import { initialState as tweetsInitialState } from '../../reducers/tweetsReducer';

describe('index pages', () => {
    describe('render()', () => {
        const initialState = {
            tweetsReducer: tweetsInitialState
        }
        const mockStore = configureStore()
        let store

        it('renders the component', () => {
            store = mockStore(initialState)
            const { firstChild } = render(<Provider store={store}><App /></Provider>)
            expect(firstChild).toMatchSnapshot()
        })
    })
});