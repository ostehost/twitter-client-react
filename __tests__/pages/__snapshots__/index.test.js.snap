// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index pages render() renders the component 1`] = `undefined`;
